x-pack/plugins/lens/public/editor_frame_service/editor_frame/state_helpers.ts

```diff
@@ -134,7 +134,7 @@ export const validateDatasourceAndVisualization = (
     ? currentVisualization?.getErrorMessages(currentVisualizationState, frameAPI)
     : undefined;
 
-  if (datasourceValidationErrors || visualizationValidationErrors) {
+  if (datasourceValidationErrors?.length || visualizationValidationErrors?.length) {
```

Comment:
I wonder if it makes sense to get rid of the `undefined` here completely, and just work with array. I cannot foresee particular differences between the `no current visualization/datasource available` case and the `no error` one in this context.

Reply:
I considered making this change in this PR, but decided not to because it actually affects the signature of all of the visualizations.